CMakeLists.txt

```diff
@@ -61,6 +61,12 @@ set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 
+execute_process(
+  COMMAND conan create . scipp/stable
+  WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/.conan-recipes/llnl-units"
+                    COMMAND_ECHO STDOUT
+)
```

Comment:
Calling `conan create` uses the users default profile. In may case, that profile specifies gcc10 but I set the CMake compiler to gcc11. This causes a build error of the `units` package:
```
CMake Error at /home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/76fffa8f7bff83f70caa3401effcaf83a365605e/conanbuildinfo.cmake:402 (message):
  Detected a mismatch for the compiler version between your conan profile
  settings and CMake:

  Compiler version specified in your conan profile: 10

  Compiler version detected in CMake: 11.1

  Please check your conan profile settings (conan profile show
  [default|your_profile_name])

  P.S.  You may set CONAN_DISABLE_CHECK_COMPILER CMake variable in order to
  disable this check.
Call Stack (most recent call first):
  /home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/76fffa8f7bff83f70caa3401effcaf83a365605e/conanbuildinfo.cmake:486 (conan_error_compiler_version)
  /home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/76fffa8f7bff83f70caa3401effcaf83a365605e/conanbuildinfo.cmake:591 (check_compiler_version)
  /home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/76fffa8f7bff83f70caa3401effcaf83a365605e/conanbuildinfo.cmake:141 (conan_check_compiler)
  CMakeLists.txt:25 (conan_basic_setup)


-- Configuring incomplete, errors occurred!
See also "/home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/76fffa8f7bff83f70caa3401effcaf83a365605e/CMakeFiles/CMakeOutput.log".
LLNL-Units/0.5.0@scipp/stable: 
LLNL-Units/0.5.0@scipp/stable: ERROR: Package '76fffa8f7bff83f70caa3401effcaf83a365605e' build failed
LLNL-Units/0.5.0@scipp/stable: WARN: Build folder /home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/76fffa8f7bff83f70caa3401effcaf83a365605e
ERROR: LLNL-Units/0.5.0@scipp/stable: Error in build() method, line 63
        cmake.configure(source_folder="units")
        ConanException: Error 1 while executing cd '/home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/76fffa8f7bff83f70caa3401effcaf83a365605e' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Release" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="gcc" -DCONAN_COMPILER_VERSION="10" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libstdc++11" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/package/76fffa8f7bff83f70caa3401effcaf83a365605e" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -DUNITS_ENABLE_TESTS="OFF" -DUNITS_BASE_TYPE="uint32_t" -DCMAKE_CXX_STANDARD="17" -Wno-dev '/home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/76fffa8f7bff83f70caa3401effcaf83a365605e/units'
-- Conan: checking conan executable
-- Conan: Found program /home/jl/Software/Miniconda3/envs/scipp-2/bin/conan
-- Conan: Version found Conan version 1.40.2

-- Conan executing: /home/jl/Software/Miniconda3/envs/scipp-2/bin/conan install /home/jl/Work/scipp-2 --build outdated --settings build_type=Debug --settings compiler=gcc --settings compiler.version=11 --settings compiler.libcxx=libstdc++11
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

benchmark/1.5.3: Package is up to date
boost/1.76.0: Package is up to date
eigen/3.3.9: Package is up to date
gtest/1.11.0: Package is up to date
pybind11/2.6.2: Package is up to date
conanfile.txt: Installing package
Requirements
    LLNL-Units/0.5.0@scipp/stable from local cache - Cache
    benchmark/1.5.3 from 'conancenter' - Cache
    boost/1.76.0 from 'conancenter' - Cache
    eigen/3.3.9 from 'conancenter' - Cache
    gtest/1.11.0 from 'conancenter' - Cache
    pybind11/2.6.2 from 'conancenter' - Cache
Packages
    LLNL-Units/0.5.0@scipp/stable:076baf79d4fefec67e834b4f4556482e7f2d4f39 - Build
    benchmark/1.5.3:86f95006b327ea67e13d6e149353d507c9f7cb77 - Cache
    boost/1.76.0:524ea35a8120baabdde02483add58d81bf541327 - Cache
    eigen/3.3.9:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    gtest/1.11.0:63868df56b76903d4ad40ecbd5b2e8238cee50c9 - Cache
    pybind11/2.6.2:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache

Installing (downloading, building) binaries...
LLNL-Units/0.5.0@scipp/stable: Copying sources to build folder
LLNL-Units/0.5.0@scipp/stable: Building your package in /home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/076baf79d4fefec67e834b4f4556482e7f2d4f39
LLNL-Units/0.5.0@scipp/stable: Generator cmake created conanbuildinfo.cmake
LLNL-Units/0.5.0@scipp/stable: Aggregating env generators
LLNL-Units/0.5.0@scipp/stable: Calling build()
-- The CXX compiler identification is GNU 11.1.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/jl/Software/Miniconda3/envs/scipp-2/bin/x86_64-conda-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Conan: Adjusting fPIC flag (ON)
-- Conan: Compiler GCC>=5, checking major version 11
-- Conan: Checking correct version: 11
-- Conan: C++ stdlib: libstdc++11
-- setting up for GNU on Linux-5.10.61-1-MANJARO
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY


-- Build files have been written to: /home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/076baf79d4fefec67e834b4f4556482e7f2d4f39
[ 80%] Building CXX object units/CMakeFiles/units.dir/r20_conv.cpp.o
[ 80%] Building CXX object units/CMakeFiles/units.dir/units.cpp.o
[ 80%] Building CXX object units/CMakeFiles/units.dir/x12_conv.cpp.o
[ 80%] Building CXX object units/CMakeFiles/units.dir/commodities.cpp.o
[100%] Linking CXX static library ../lib/libunits.a
[100%] Built target units
LLNL-Units/0.5.0@scipp/stable: Package '076baf79d4fefec67e834b4f4556482e7f2d4f39' built
LLNL-Units/0.5.0@scipp/stable: Build folder /home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/build/076baf79d4fefec67e834b4f4556482e7f2d4f39
LLNL-Units/0.5.0@scipp/stable: Generated conaninfo.txt
LLNL-Units/0.5.0@scipp/stable: Generated conanbuildinfo.txt
LLNL-Units/0.5.0@scipp/stable: Generating the package
LLNL-Units/0.5.0@scipp/stable: Package folder /home/jl/.conan/data/LLNL-Units/0.5.0/scipp/stable/package/076baf79d4fefec67e834b4f4556482e7f2d4f39
LLNL-Units/0.5.0@scipp/stable: Calling package()
LLNL-Units/0.5.0@scipp/stable package(): Packaged 4 '.hpp' files: units_util.hpp, units_decl.hpp, units.hpp, unit_definitions.hpp
LLNL-Units/0.5.0@scipp/stable package(): Packaged 1 '.a' file: libunits.a
LLNL-Units/0.5.0@scipp/stable: Package '076baf79d4fefec67e834b4f4556482e7f2d4f39' created
LLNL-Units/0.5.0@scipp/stable: Created package revision b85faa4122ff3864f6a18c01b9823ea4
```

But CMake continues regardless and invokes conan later to install `units`. I don't know where it gets the recipe from, but I think it uses the local one. Can you make sure this is actually the case and conan does not use an incorrect package version.

Reply:
So I would need to somehow configure which profile it uses?